### PR TITLE
Reintroduce AWS Leasing

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/assign @wking

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@wking failures are all flakes.

```
$ oc logs -n ci-op-0nl0fbn4 e2e-aws-upi -c lease
[INFO] Acquiring a lease ...
[INFO] Lease acquired!
[INFO] Leased resource: {"type":"aws-quota-slice","name":"cc65a053-4fa0-451e-9969-164180476bbf","state":"leased","owner":"ci-op-0nl0fbn4-54351","lastupdate":"2019-08-06T20:12:29.664305893Z","userdata":{}}
[INFO] Sending heartbeats to retain the lease...
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
heartbeat sent for resource "cc65a053-4fa0-451e-9969-164180476bbf"
Another process exited
[INFO] Releasing the lease on resouce cc65a053-4fa0-451e-9969-164180476bbf...
released resource "cc65a053-4fa0-451e-9969-164180476bbf"
```

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 1]
```diff
@@ -132,7 +132,7 @@ objects:
         touch /tmp/shared/leased
 
         function release() {
-            local resource_name; resource_name="$( jq .name <<<"${resource}" )"
+            local resource_name; resource_name="$( jq .name <<<"${resource}" --raw-output )"
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: nit: Can we use `jq --raw-output .name <<<"${resource}"`?  I'm not used to seeing the option way off on the right after a positional argument and here-string.

Comment (stevekuznetsov) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: lol, yeah -- that's insane that it even works

File: ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml [hunk 2]
```diff
@@ -135,7 +135,7 @@ objects:
         echo "[INFO] Leased resource: ${resource}"
 
         function release() {
-            local resource_name; resource_name="$( jq .name <<<"${resource}" --raw-output )"
+            local resource_name; resource_name="$( jq .name --raw-output <<<"${resource}" )"
```
Comment (wking) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-console.yaml]: still has the option to the right of the `.name` positional argument, but whatever ;)

/lgtm

**Pull request merged**